api-channel/src/main/java/com/synopsys/integration/alert/api/channel/DistributionEventHandler.java

```diff
@@ -25,19 +28,24 @@
     private final DistributionChannel<D> channel;
     private final JobDetailsAccessor<D> jobDetailsAccessor;
     private final ProcessingAuditAccessor auditAccessor;
+    private final JobAccessor jobAccessor;
 
-    public DistributionEventHandler(DistributionChannel<D> channel, JobDetailsAccessor<D> jobDetailsAccessor, ProcessingAuditAccessor auditAccessor) {
+    public DistributionEventHandler(DistributionChannel<D> channel, JobDetailsAccessor<D> jobDetailsAccessor, ProcessingAuditAccessor auditAccessor, JobAccessor jobAccessor) {
         this.channel = channel;
         this.jobDetailsAccessor = jobDetailsAccessor;
         this.auditAccessor = auditAccessor;
+        this.jobAccessor = jobAccessor;
     }
 
     @Override
     public final void handle(DistributionEvent event) {
-        Optional<D> details = jobDetailsAccessor.retrieveDetails(event.getJobId());
-        if (details.isPresent()) {
+        Optional<D> detailsOptional = jobDetailsAccessor.retrieveDetails(event.getJobId());
+        if (detailsOptional.isPresent()) {
+            D details = detailsOptional.get();
+            String jobName = jobAccessor.getJobById(details.getJobId()).map(DistributionJobModel::getName).orElse("UNKNOWN");
+            NamedDistributionJobDetailsModel<D> namedDistributionJobDetailsModel = new NamedDistributionJobDetailsModel<>(details, jobName);
```

Comment:
I really struggle with making two DB calls here. Is there no chance that `JobDetailsAccessor::retrieveDetails` could return a `NamedDistributionJobDetailsModel<D>`? The entire point of `JobDetailsAccessor` is so that we don't have to retrieve the entire job.

Reply:
That was my original approach until I realized that there's a JobDetailsAccessor object for each specific channel which only queries the jobs specific table (Which doesn't have my data). So I'd need to join tables or do 2 queries still. this seemed like the best way to make the fewest amount of changes. Is there are better spot to retrieve this data than something like DefaultAzureBoardsJobDetailsAccessor?

Reply:
Those accessors are designed for this class. If this class needs new things, those accessors should change IMO.